typing/env.ml

```diff
@@ -1803,6 +1801,14 @@ and add_extension ~check id ext env =
   store_extension ~check id addr ext env
 
 and add_module_declaration ?(arg=false) ~check id presence md env =
+  let check =
+    if not check then
+      None
+    else if arg && is_in_signature env then
+      Some (fun s -> Warnings.Unused_functor_parameter s)
+    else
+      Some (fun s -> Warnings.Unused_module s)
+  in
```

Comment:
Since `add_module_declaration` is the only user of the check flag, would it not be more straightforward to move the unused warning code from `store_module` here?

Reply:
That'd be inconsistent with `add_type / store_type`, `add_value / store_value`.

Reply:
This sounds like a good reason.